components/UI/navbar.tsx

```diff
@@ -90,9 +90,16 @@ const Navbar: FunctionComponent = () => {
   }, [isConnected]);
 
   const connectWallet = async (connector: Connector) => {
-    await connectAsync({ connector });
-    localStorage.setItem("SID-connectedWallet", connector.id);
-    localStorage.setItem("SID-lastUsedConnector", connector.id);
+    try {
+      await connectAsync({ connector });
+      localStorage.setItem("SID-connectedWallet", connector.id);
+      localStorage.setItem("SID-lastUsedConnector", connector.id);
+    } catch (e) {
+      // Restart the connection if there is an error except if the user has rejected the connection
```

Comment:
Why do we need to restart the connection ? It should not fail if the connector is `available`

Reply:
It throws this weird error sometimes (33% of the time), even with autoconnect which excplictly checks if it's available. Trying again to connect solves this issue (which I guess I a bug on starknetkit or starknetreact or is a problem of incompatibility between lib versions).
![image](https://github.com/starknet-id/app.starknet.id/assets/60229704/9a82a78b-1ac8-4ba0-9b56-a7ef3355b41a)

![image](https://github.com/starknet-id/app.starknet.id/assets/60229704/c33607e9-cbb7-4b2a-92f9-f98e436d17e4)

Reply:
Ah ok so maybe it takes too much time to inject itself on mobile. Let's leave it like this for now then, and hopefully we'll be able to remove it at some point.